sdk/core/Azure.Core/src/ProcessingContext.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.Core
+{
+    /// <summary>
+    /// Contains information related to the processing of the <see cref="HttpMessage"/>.
+    /// </summary>
+    public class ProcessingContext
+    {
+        /// <summary>
+        /// The start time of the first try of the request.
+        /// </summary>
+        public DateTimeOffset OperationStartTime { get; internal set; }
+
+        /// <summary>
+        /// The attempt number for the current attempt.
+        /// </summary>
+        public int AttemptNumber { get; internal set; }
+
+        /// <summary>
+        /// The exception that occurred on the previous attempt, if any.
+        /// </summary>
+        public Exception? LastException { get; internal set; }
```

Comment:
If someone implements their own retry policy, should they be able to modify any of these values?

Reply:
Hmm, good question. Right now it is read-only, but we could consider making these settable so that someone that is fully overriding the RetryPolicy can choose to set these properties.